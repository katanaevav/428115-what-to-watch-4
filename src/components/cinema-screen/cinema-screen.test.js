import React from "react";
import renderer from "react-test-renderer";
import CinemaScreen from "./cinema-screen.jsx";

const Movie = {
  id: 0,
  title: `Aviator`,
  smallPoster: `img/aviator.jpg`,
  genre: `Drama`,
  year: 2004,
  runTime: 180,
  bigPoster: `img/aviator.jpg`,
  cover: `img/aviator.jpg`,
  ratingScore: 7.5,
  ratingCount: 121697,
  isFavorite: false,
  backgroundColor: `#AABBCC`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  directors: [`Мартин Скорсезе`],
  starrings: [`Леонардо ДиКаприо`, `Кейт Бланшетт`, `Мэтт Росс`, `Джон Си Райли`, `Алан Алда`, `Кейт Бекинсейл`, `Алек Болдуин`, `Иэн Холм`, `Адам Скотт`, `Дэнни Хьюстон`],
  descriptions: [
    `Получив от отца небольшую фабрику, Говард Хьюз превратил ее в гигантское, фантастически прибыльное предприятие. Став владельцем огромной кинокомпании, он снял самый дорогой для своего времени фильм и покорил сердца прелестнейших голливудских актрис.`,
    `Ему принадлежали самые престижные казино Лас-Вегаса и он установил рекорд скоростных полетов, приобрел вторую по величине коммерческую авиакомпанию…`,
    `Деньги жгут сердце Хьюза, они не дают ему покоя, а душа его рвется ввысь. Только там, на высоте нескольких тысяч метров он счастлив по-настоящему. Только там, где все решает лишь мастерство пилота и Бог, ничто не ценится так дорого, как верность и честь.`
  ],
};

it(`Render cinema screen`, () => {
  const tree = renderer.create(
      <CinemaScreen
        movie = {Movie}
        renderPlayer = {() => {}}
        renderProgress = {() => {}}
        renderPlayButton = {() => {}}
        onFullScreenButtonClick = {() => {}}
        onExitVideoPlayer = {() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
