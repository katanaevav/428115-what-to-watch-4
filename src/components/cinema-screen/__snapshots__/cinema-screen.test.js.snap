// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render cinema screen 1`] = `
<div
  className="player"
>
  <button
    className="player__exit"
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__name"
      >
        Aviator
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="/sprite.svg#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
